validphys2/src/validphys/utils.py

```diff
@@ -5,12 +5,51 @@
 @author: Zahari Kassabov
 """
 import contextlib
+import functools
 import pathlib
 import shutil
 import tempfile
+from typing import Any, Sequence, Mapping
 
 import numpy as np
 from validobj import ValidationError, parse_input
+from frozendict import frozendict
+
+
+# Since typing.Hashable doesn't check recursively you actually
+# have to try hashing it.
```

Comment:
This I don't really understand, what cases would we miss by doing `isinstance(obj,Hashable)`?

Reply:
I agree, and I prefer the type check you propose. Will implement it

Reply:
It works just as well, see 19771a027d0dacfc966c339180680bdeda06954c